src/stacked_borrows.rs

```diff
@@ -621,6 +694,21 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
         if size == Size::ZERO {
             // Nothing to do for zero-sized accesses.
+            // But we still create a tag for them so that it's clear if a user misuses a pointer
```

Comment:
This is confusing, since the code here does not create a tag.

Also I think the code duplication for `CreatedPointerTag` should be avoided.